frame/header-mmr/src/lib.rs

```diff
@@ -62,37 +62,62 @@ mod tests;
 
 #[frame_support::pallet]
 pub mod pallet {
+	pub mod types {
+		/// The type use for indexing a node
+		pub type NodeIndex = u64;
+	}
+	pub use types::*;
+
 	// --- crates.io ---
 	#[cfg(feature = "std")]
 	use serde::Serialize;
 	// --- github.com ---
 	use mmr::{Error, MMRStore, Merge, Result as MMRResult, MMR};
-	// --- substrate ---
-	use frame_support::pallet_prelude::*;
+	// --- paritytech ---
+	use frame_support::{pallet_prelude::*, weights::Weight};
 	use frame_system::pallet_prelude::*;
 	use sp_runtime::{
 		generic::{DigestItem, OpaqueDigestItemId},
 		traits::{Hash, Header, SaturatedConversion},
 	};
 	use sp_std::prelude::*;
 	// --- darwinia ---
+	use crate::weights::WeightInfo;
 	use darwinia_header_mmr_rpc_runtime_api::{Proof, RuntimeDispatchInfo};
 	use darwinia_relay_primitives::MMR as MMRT;
 
-	pub const PARENT_MMR_ROOT_LOG_ID: [u8; 4] = *b"MMRR";
+	// ? Useless const
+	// ? commented by Xavier
+	/// The prefix of [`MerkleMountainRangeRootLog`]
+	pub const LOG_PREFIX: [u8; 4] = *b"MMRR";
 
 	#[pallet::config]
-	pub trait Config: frame_system::Config {}
+	pub trait Config: frame_system::Config {
+		/// The offchain-indexing prefix
+		const INDEXING_PREFIX: &'static [u8];
+
+		type WeightInfo: WeightInfo;
+	}
 
-	/// The MMR size and length of the mmr node list
+	/// Size of the MMR
 	#[pallet::storage]
-	#[pallet::getter(fn mmr_counter)]
-	pub type MMRCounter<T> = StorageValue<_, u64, ValueQuery>;
+	#[pallet::getter(fn mmr_size)]
+	pub type MmrSize<T> = StorageValue<_, NodeIndex, ValueQuery>;
 
 	/// MMR struct of the previous blocks, from first(genesis) to parent hash.
 	#[pallet::storage]
 	#[pallet::getter(fn mmr_node_list)]
-	pub type MMRNodeList<T: Config> = StorageMap<_, Identity, u64, T::Hash, OptionQuery>;
+	pub type MMRNodeList<T: Config> = StorageMap<_, Identity, NodeIndex, T::Hash, OptionQuery>;
+
+	/// Peaks of the MMR
+	#[pallet::storage]
+	#[pallet::getter(fn peak_of)]
+	pub type Peaks<T: Config> = StorageMap<_, Identity, NodeIndex, T::Hash, OptionQuery>;
```

Comment:
Do not suggest using new storage in this stage, as the pruning might take several days, we can do renaming in later refactor pr, Please keep this pr as simple as possible, one thing at a time.